qupulse/_program/_loop.py

```diff
@@ -547,12 +559,19 @@ def _make_compatible(program: Loop, min_len: int, quantum: int, sample_rate: Tim
 
 
 def make_compatible(program: Loop, minimal_waveform_length: int, waveform_quantum: int, sample_rate: TimeType):
+    """ check program for compatibility, make it compatible if necessary and  possible"""
+
     comp_level = _is_compatible(program,
                                 min_len=minimal_waveform_length,
                                 quantum=waveform_quantum,
                                 sample_rate=sample_rate)
-    if comp_level == _CompatibilityLevel.incompatible:
-        raise ValueError('The program cannot be made compatible to restrictions')
+    if comp_level == _CompatibilityLevel.incompatible_fraction:
+        raise ValueError('The program duration in samples is not an integer')
+    if comp_level == _CompatibilityLevel.incompatible_too_short:
+        raise ValueError('The program is too short to be a valid waveform')
+    if comp_level == _CompatibilityLevel.incompatible_quantum:
+        raise ValueError('The program duration is not a multiple of quantum')
```

Comment:
Maybe we can even give the two values `duration_in_samples` and (`quantum` / `minimum_length`) where appropriate